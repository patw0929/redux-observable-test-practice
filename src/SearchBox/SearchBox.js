import React from 'react';

const SearchBox = ({ value, onChange }) => (
  <div className="form-group">
    <label htmlFor="search-input">
      Search GitHub repos:
    </label>

    <input
      type="text"
      id="search-input"
      className="form-control"
      value={value}
      onChange={e => onChange.call(this, e.target.value)}
      placeholder="Search"
    />
  </div>
);

export default SearchBox;
