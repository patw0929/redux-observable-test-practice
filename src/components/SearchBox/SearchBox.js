import React from 'react';
import DebounceInput from '../../hoc/DebounceInput';

const SearchBox = ({ onChange, props$ }) => (
  <div className="form-group">
    <label htmlFor="search-input">
      Search GitHub repos:
    </label>

    <input
      type="text"
      id="search-input"
      className="form-control"
      onChange={e => props$.next(e.target.value)}
      placeholder="Search"
    />
  </div>
);

export default DebounceInput(
  SearchBox
);
