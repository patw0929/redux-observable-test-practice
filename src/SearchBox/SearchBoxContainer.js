import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import SearchBox from './SearchBox';
import { search } from '../redux/modules/searchByKeyword';

const mapDispatchToProps = dispatch => {
  return {
    onChange: bindActionCreators(search, dispatch),
  };
};

export default connect(null, mapDispatchToProps)(SearchBox);
