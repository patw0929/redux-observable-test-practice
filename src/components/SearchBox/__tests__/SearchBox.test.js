import React from 'react';
import { shallow } from 'enzyme';
import SearchBox from '../SearchBox';

describe('SearchBox', () => {
  it('renders correct as snapshot', () => {
    const subject = shallow(<SearchBox props$={jest.fn} />);

    expect(subject.html()).toMatchSnapshot();
  });
});
