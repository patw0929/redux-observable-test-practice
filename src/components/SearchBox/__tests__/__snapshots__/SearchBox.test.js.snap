// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox renders correct as snapshot 1`] = `"<div class=\\"form-group\\"><label for=\\"search-input\\">Search GitHub repos:</label><input type=\\"text\\" id=\\"search-input\\" class=\\"form-control\\" placeholder=\\"Search\\"/></div>"`;
